### DISCUSSION TITLE
Use Fetch in Firestore v8

### DISCUSSION THREAD

**Discussion by maucaro** [firebase/firebase-js-sdk#6108] [General]
I see that the Firestore v9 SDK was changed to use fetch instead of XHR per [this](https://github.com/firebase/firebase-js-sdk/pull/4939). 

Can the same be done for v8? I'd like to use Firestore in the context of a Service Worker. I was able to get this working using v9 but unfortunately the browser support for ES modules in service workers is still limited (Chrome and Edge since 91; Safari since 15; Firefox none; see [this](https://github.com/mozilla/standards-positions/issues/499)).

**Comment by hsubox76** [comment]
Unfortunately we are not updating v8 with the rare exception of major security issues.

I think even in v9 you should be able to use the "compat" bundles from the CDN to work around this, e.g.

```
https://www.gstatic.com/firebasejs/9.3.0/firebase-app-compat.js
https://www.gstatic.com/firebasejs/9.3.0/firebase-firestore-compat.js
```

These should be in UMD format. That might be a possible solution?

**Reply by maucaro** [reply]
Thank you @hsubox76 ; I will give this suggested approach a try.
